### PR TITLE
Add config/docker command

### PR DESCRIPTION
**Pull request by jasonhu5**
Please see https://github.com/duckietown/duckietown-shell-commands/pull/362 for the description and test plan

### CODE REVIEW

**Comment by jasonhu5** [comment]
The extra file `duckiebot/demo/command.py` contains the changes in https://github.com/duckietown/duckietown-shell-commands/pull/358 that has probably been accidentally reverted when rebasing daffy-staging.

**Comment by jasonhu5** [comment]
### Summary

#### Added `dts config docker` command
* with these sub-commands:
Screenshot 2023-02-11 at 17 57 03

#### Tested `dts config docker info` with empty entry and multi entry local config file
* `--help`
Screenshot 2023-02-11 at 17 56 21

* no docker entry
Screenshot 2023-02-11 at 17 56 02

* multiple docker entries
Screenshot 2023-02-11 at 17 54 52

#### Test plans and usage for `dts config docker set` are stated in the initial comment

(Internal: DTSW-2210)

File: config/docker/set/command.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        
+        if username is None and password is None:
+            print(DTCommand.usage)
+            exit(4)
+
+        if username is not None:
+            shell.shell_config.docker_username = username
+            shell.shell_config.docker_credentials[server]["username"] = username
+        if password is not None:
+            shell.shell_config.docker_password = password
```
Comment (afdaniele) [inline config/docker/set/command.py]: This is deprecated, no need to set.

File: config/docker/set/command.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        if server not in shell.shell_config.docker_credentials:
+            shell.shell_config.docker_credentials[server] = {}
+        
+        if username is None and password is None:
+            print(DTCommand.usage)
+            exit(4)
+
+        if username is not None:
+            shell.shell_config.docker_username = username
```
Comment (afdaniele) [inline config/docker/set/command.py]: This is deprecated, no need to set.

File: config/docker/set/command.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+from dt_shell import DTCommandAbs, dtslogger, DTShell
+
+
+class DTCommand(DTCommandAbs):
+    help = "Configure docker registry credentials"
+
+    usage = f"""
+Usage:
+
+    dts config docker --docker-username <your-docker-username> --docker-password <your-docker-access-token> [--docker-registry <optional-specific-registry>]
```
Comment (afdaniele) [inline config/docker/set/command.py]: Replace `--docker-username` with `--username`, `--docker-password` with `--password`, and `--docker-registry` with an optional positional argument called `registry`.

**Review by afdaniele — CHANGES_REQUESTED** [review]

File: config/docker/set/command.py (hunk 3 above)
Comment (jasonhu5) [reply config/docker/set/command.py]: For the same reason, to resemble `docker login` ([format](https://docs.docker.com/engine/reference/commandline/login/)), I named the `registry` positional argument as `server` in the new commit

**Comment by jasonhu5** [comment]
After the adjustment, the `--help` output is:
image

**Review by afdaniele — APPROVED** [review]

**Pull request merged**